addons/sale_coupon/models/sale_order.py

```diff
@@ -22,12 +22,21 @@ def _compute_reward_total(self):
         for order in self:
             order.reward_amount = sum([line.price_subtotal for line in order._get_reward_lines()])
 
+    def _get_no_effect_delivery_lines(self):
+        self.ensure_one()
+        # Do not count shipping
+        lines = self.order_line.filtered(lambda line: line.is_delivery)
+        return lines
+
     def _get_no_effect_on_threshold_lines(self):
         self.ensure_one()
         lines = self.env['sale.order.line']
         # Do not count already applied promo_code discount; Do not substract itself
         if self.code_promo_program_id and self.code_promo_program_id.reward_type == 'discount':
-            lines = self.order_line.filtered(lambda l: l.product_id == self.code_promo_program_id.discount_line_product_id)
+            lines |= self.order_line.filtered(lambda l: l.product_id == self.code_promo_program_id.discount_line_product_id)
```

Comment:
Comment 1: Imo, `_get_no_effect_on_threshold_lines()` should remain as it was in sale_coupon_delivery and return delivery + free delivery, so this method will return the lines that always have no effect on the check on minimum threshold amount requirement.
In sale_coupon, `_get_no_effect_on_threshold_lines()` could just return empty recordset. 
That would also mean if someone has a custom module inserting lines in that method, it will still work as before.